tests/test_juicebox_assembly_purger.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python
+'''
+Brad Nelson
+May 30, 2019
+Phase Genomics
+
+tests/test_juicebox_assembly_purger.py
+
+This file contains unit tests for functions of the juicebox_assembly_purger.py script.
+
+Copyright 2018, Phase Genomics Inc. All rights reserved.
+
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU Affero General Public License as
+published by the Free Software Foundation, either version 3 of the
+License, or (at your option) any later version.
+
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with this program.  If not, see https://www.gnu.org/licenses/agpl-3.0.en.html
+'''
+
+import os
+import filecmp
+import unittest
+
+from juicebox_scripts.juicebox_assembly_purger import filter_assembly, get_exclude
+
+class JuiceboxPurgerTestCase(unittest.TestCase):
+    def setUp(self):
+        self.collateral = os.path.dirname(__file__) + "/collateral/"
+        self.input_assembly = self.collateral + "test_inputs/test_scaffolds.assembly"
+        self.input_exclude_file = self.collateral + "test_inputs/test_exclude.txt"
+        self.output_dir = self.collateral + "test_outputs/"
+        self.expected_output_dir = self.collateral + "expected_test_outputs/"
+
+        if not os.path.exists(self.output_dir):
+            os.makedirs(self.output_dir)
+
+    def tearDown(self):
+        outfiles = ["test_filter_assembly_null.assembly",
+                    "test_filter_assembly_minus_one.assembly",
+                    "test_filter_assembly_minus_three.assembly",
+                    "test_filter_assembly_missing_contig.assembly"]
+        for outfile in outfiles:
+            outfile = self.output_dir + outfile
+            if os.path.exists(outfile):
+                os.remove(outfile)
```

Comment:
Should we delete self.output_dir too? I don't really have a preference but just asking for thoughts

Reply:
I don't think removing the outputdir as part of the `tearDown` method is a good idea, since that method is called after every test and could put other tests in a bad state. Instead, I'm using the `tearDownClass` method to remove the directory once at the end.

Reply:
That seems like a great solution. Didn't know about those Class methods!